src/misc-utils.test.ts

```diff
@@ -0,0 +1,63 @@
+import { writeFile } from '@metamask/utils/node';
+import fs from 'fs';
+import path from 'path';
+
+import { getEntryStats } from './misc-utils';
+import { withinSandbox } from '../tests/helpers';
+
+describe('getEntryStats', () => {
+  describe('given a file', () => {
+    it('returns the stats for the file', async () => {
+      expect.assertions(3);
+
+      await withinSandbox(async (sandbox) => {
+        const filePath = path.join(sandbox.directoryPath, 'nonexistent.file');
+        await writeFile(filePath, '');
+
+        const stats = await getEntryStats(filePath);
+
+        expect(stats).toHaveProperty('atime');
+        expect(stats).toHaveProperty('ctime');
+        expect(stats).toHaveProperty('mtime');
+      });
+    });
+
+    it('returns null if the file does not exist', async () => {
+      expect.assertions(1);
+
+      await withinSandbox(async (sandbox) => {
+        const filePath = path.join(sandbox.directoryPath, 'nonexistent.file');
+
+        expect(await getEntryStats(filePath)).toBeNull();
+      });
+    });
+
+    it('re-throws a wrapped version of any other error that occurs, assigning it the same code and giving it a stack', async () => {
+      expect.assertions(1);
+
+      await withinSandbox(async (sandbox) => {
+        const filePath = path.join(sandbox.directoryPath, 'nonexistent.file');
+        await writeFile(filePath, '');
+        try {
+          // Make sandbox root directory non-executable
+          await fs.promises.chmod(sandbox.directoryPath, 0o600);
+
+          await expect(getEntryStats(filePath)).rejects.toThrow(
+            expect.objectContaining({
+              message: `Could not get stats for file or directory '${filePath}'`,
+              code: 'EACCES',
+              stack: expect.any(String),
+              cause: expect.objectContaining({
+                message: `EACCES: permission denied, stat '${filePath}'`,
+                code: 'EACCES',
+              }),
+            }),
+          );
+        } finally {
+          // Make sandbox root directory executable again
+          await fs.promises.chmod(sandbox.directoryPath, 0o700);
```

Comment:
Nit: Is this step necessary? I thought using a sandbox removed the necessity to clean up; nothing will use this directory again after this test

Reply:
If I remove this line, the `chmod` above seems to prohibit the sandbox from removing itself. I tried to remove the sandbox directory manually and got:

```
rm: /path/to/temp/directory/module-lint-tests--1699398082717/test-directory: Permission denied
rm: /path/to/temp/directory/module-lint-tests--1699398082717: Permission denied
```

Perhaps `withinSandbox` should set the `execute` permissions on the directory before deleting it.

Anyway, it's a bug that we should address in `utils`. I've added a reminder here: [16d512e](https://github.com/MetaMask/module-lint/pull/21/commits/16d512ea81e6afc5537c6d8de87407c4be8aab5d)

Reply:
Ah interesting, OK